examples/systemcalls/systemcalls.c

```diff
@@ -92,8 +137,56 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
  *   The rest of the behaviour is same as do_exec()
  *
 */
+    int fd = open(outputfile, O_WRONLY|O_TRUNC|O_CREAT, 0644);
+    if(-1 == fd) {
+        perror(" File open: ");
+        exit(-1);
+    }
+    int ret_status = true;
+    pid_t ret_val = fork();
+
+    // check for errno in fork 
+    if(-1 == ret_val) {
+        ret_status = false; 
+        perror("Fork error: ");
+    }
+    // parent operation to wait for child to turn zombie
+    else if(ret_val > 0) {
+        int status; 
+        pid_t wait_status = wait(&status);
+        if(-1 == wait_status) {
+            perror("Wait error: ");
+            ret_status = false; 
+        }
+        else {
+            // check if child called _exit to exit normally
+            if(WIFEXITED(status)) {
+                // check if child process did not exit success(failure == 1)
+                if(WEXITSTATUS(status) != 0) {
+                   ret_status = false; 
+                }
+            } else {
+                // if child did not call _exit() 
+                ret_status = false; 
```

Comment:
Good!
See discussion in class about avoiding duplication between these two very similar functions.